compiler/dcalc/interpreter.ml

```diff
@@ -188,6 +188,8 @@ let rec evaluate_operator (ctx : Ast.decl_ctx) (op : A.operator Pos.marked)
         match Pos.unmark (evaluate_operator ctx (Pos.same_pos_as (A.Binop A.Eq) op) args) with
         | A.ELit (A.LBool b) -> A.ELit (A.LBool (not b))
         | _ -> assert false (*should not happen *))
+    | A.Binop A.Concat, [ A.EArray es; A.EArray es2 ] ->
+        A.EArray (es @ es2) (* @EmileRolley NOTE: should it keep lazy evaluated or not? *)
```

Comment:
Keep it like that. `es` and `es2` already have been evaluated at this point because concat is viewed as a function, and when you call a function you always evaluate first the arguments.

Reply:
Ok I see. I will remove the comment.